### ISSUE TITLE
Find a way to make `checkAccessibility` helpers work when rendered component's first child is a Fragment

### ISSUE THREAD

**Issue by acelaya** [hypothesis/client#5690]
We have found that our accessibility tests can fail if tested component returns a Fragment as its uppermost children (see https://github.com/hypothesis/client/issues/5671).

The reason is that inside `checkAccessibility`, there's one point in which we call `wrapper.getDOMNode()`. This method cannot translate a Fragment into a native DOM node, which results in just the first Fragment's children to be returned, ignoring the rest.

At the moment we have worked around this by wrapping the component under test in another element (for example, a `div`. See https://github.com/hypothesis/client/pull/5672/files), but it would be good to find a more resilient way to address it that does not require remembering extra steps.
